### PR TITLE
Haskell adoption

### PR DESCRIPTION
**Pull request by erthalion**
Change examples and some consents from Clojure to Haskell

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -64,15 +64,14 @@ <h4>Introduction</h4>
         <script type="text/template">
           ### Expressions
 
-          ```clj
-          (+ 1 2 3)         ; 1 + 2 + 3       ; => 6
-          (println "Hello") ; println("Hello) ; => nil
-          (>= 10 5)         ; 10 >= 5         ; => true
-          (if (>= 10 5) "Yes" "No") ; => "Yes"
+          ```haskell
```
Comment (dryewo) [inline index.html:67]: I'd suggest indentation of comments

File: index.html [hunk 2]
```diff
@@ -64,15 +64,14 @@ <h4>Introduction</h4>
...
+          1 + 2 + 3       -- => 6
+          putStrLn "Hello" -- => IO ()
+          10 >= 5         -- => true
+          if 10 >= 5 then "Yes" else "No") -- => "Yes"
```
Comment (dryewo) [inline index.html]: `)` is not needed

File: index.html [hunk 3]
```diff
@@ -110,13 +109,13 @@ <h4>Introduction</h4>
 
           Some expressions have side effects:
```
Comment (dryewo) [inline index.html]: This was only relevant to Clojure. In Haskell things can be used before they are defined.

File: index.html [hunk 4]
```diff
@@ -126,15 +125,9 @@ <h4>Introduction</h4>
 
           When the source code file is loaded, expressions from it are evaluated top to bottom, one by one.
```
Comment (dryewo) [inline index.html]: This is not true for Haskell

File: index.html [hunk 5]
```diff
@@ -196,25 +189,15 @@ <h4>Introduction</h4>
...
             <tbody>
             <tr>
-              <td>Lists</td>
+              <td>Tuples</td>
               <td>`(1 2 3)`</td>
               <td>O(N) access by element number</td>
             </tr>
             <tr>
-              <td>Vectors</td>
+              <td>List</td>
```
Comment (dryewo) [inline index.html]: `Lists`?

File: index.html [hunk 6]
```diff
@@ -325,11 +300,10 @@ <h4>Introduction</h4>
...
           * `n! = n * (n-1)!`
 
-          ```clj
-          (defn factorial [n]
-            (if (<= n 1)
-              1
-              (*' n (factorial (- n 1)))))
+          ```haskell
+          factorial n =
+              if n <= 1 then 1
```
Comment (dryewo) [inline index.html:290]: maybe a new line before `then`?

File: index.html [hunk 7]
```diff
@@ -524,15 +478,12 @@ <h4>Introduction</h4>
...
+          -- (:) is used to create a list:
+          1 : 2 : [3]
 
-          ;; Or even shorter
-          '(42 "Hello" "World")
+          -- Or just
+          [1, 2, 3]
           ```
 
           **'** before the list prevents Clojure from calling `42` as a function, which would make no sense.
```
Comment (dryewo) [inline index.html]: should not mention Clojure

File: index.html (hunk 6 above)
Comment (erthalion) [reply index.html:290]: I don't think it's necessary. An ideal form for Haskell is an equation, like:

```haskell
factorial 0 = 1
factorial n = n * factorial (n - 1)
```
But there is no slide for pattern matching yet, and the current form is closest to that.

**Pull request closed without merging**